src/Raven.Server/Documents/Sharding/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForDeleteOngoingTask.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Server.Documents.Handlers.Processors.OngoingTasks;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.OngoingTasks
+{
+    internal class ShardedOngoingTasksHandlerProcessorForDeleteOngoingTask : AbstractOngoingTasksHandlerProcessorForDeleteOngoingTask<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedOngoingTasksHandlerProcessorForDeleteOngoingTask([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask RaiseNotificationForSubscriptionTaskRemoval()
+        {
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Aviv, DevelopmentHelper.Severity.Normal, "how should we handle other nodes? ");
+            var tasks = RequestHandler.ServerStore.DatabasesLandlord.TryGetOrCreateShardedResourcesStore(RequestHandler.DatabaseName);
+            foreach (var task in tasks)
```

Comment:
This is weird? This only handles local server and all shards on it. What are we trying to do here?

Reply:
right, this only handles local server - that's why the todo comment is there
I'm trying to invoke the `OnRemoveTask` Action in `SubscriptionStorage`
see here :
- https://github.com/ravendb/ravendb/blob/v5.3/src/Raven.Server/Web/System/OngoingTasksHandler.cs#L1421-#L1424
- https://github.com/ravendb/ravendb/blob/f8f640b39728bd8b41c82a3e5241e032cb52eac0/src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs#L726